### ISSUE TITLE
BoxGeometry: Consider Size Array

### ISSUE THREAD

**Issue by philmaker1** [xeolabs/xeoengine#32]
Consider using key "size" with a size array instead of "xSize", "ySize", "zSize".

**Comment by xeolabs** [comment]
Yes I was wondering whether I should be doing that - only reason I used the non-array form was for consistency with other scalar properties, for example on XEO.PlaneGeometry, which has properties like ````xSegments```` and ```ySegments``` (see screenshot from XEO.PlaneGeometry API docs below):

![screenshot from 2016-01-13 10 50 03](https://cloud.githubusercontent.com/assets/83100/12290711/8cf2353a-b9e3-11e5-805d-698939703c04.png)

Would be interested though what use cases you've found for having arrays, might be worth breaking that consistency if they're compelling..

**Comment by philmaker1** [comment]
I was really comparing to the xyz array of XEO.Transforms. But overall I like using objects and properties better than arrays because they are more descriptive - for example: size.x, segments.x, xyz.x.
